cfdm/data/abstract/compressedarray.py

```diff
@@ -400,6 +400,28 @@ def conformed_data(self):
         """
         return {"data": self.source().copy()}
 
+    def get_filenames(self):
+        """Return the names of any files containing the compressed data.
+
+        .. versionadded:: (cfdm) 1.10.0.2
+
+        :Returns:
+
+            `set`
+                The file names in normalised, absolute form. If the
+                data are all in memory then an empty `set` is
+                returned.
+
+        """
+        data = self._get_compressed_Array(None)
+        if data is None:
+            return set()
+
+        try:
+            return data.get_filenames()
+        except AttributeError:
+            return set()
```

Comment:
It looks like only `get_filename` can produce an `AttributeError` now, with the plural equivalents all catching those and returning the empty set (and I double checked this includes `data.get_filenames`), so:

```suggestion
        return data.get_filenames()
```

is sufficient, unless I am missing something?

Reply:
Hi Sadie, The `AttributeError` here is to catch the case that the underlying compressed array does not support the `get_filenames` API (who knows what something like cf-python might put there :)), so I think that we still need it ...

Reply:
Oh fair, that makes sense. Thanks for clarifying, marking as resolved.